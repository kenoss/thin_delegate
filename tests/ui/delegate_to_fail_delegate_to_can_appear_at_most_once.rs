#[thin_delegate::external_trait_def]
mod __external_trait_def {
    #[thin_delegate::register]
    pub trait AsRef<T: ?Sized> {
        /// Converts this type into a shared reference of the (usually inferred) input type.
        #[stable(feature = "rust1", since = "1.0.0")]
        fn as_ref(&self) -> &T;
    }
}

#[thin_delegate::register]
enum Hoge {
    A(String),
    #[delegate_to(x => &x.0)]
    #[delegate_to(x => &x.0)]
    B((String, u8)),
}

#[thin_delegate::derive_delegate(external_trait_def = __external_trait_def)]
impl AsRef<str> for Hoge {}

fn main() {}
