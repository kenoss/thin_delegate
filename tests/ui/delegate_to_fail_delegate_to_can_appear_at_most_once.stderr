error: #[delegate_to(...)] can appear at most once
  --> tests/ui/delegate_to_fail_delegate_to_can_appear_at_most_once.rs:15:5
   |
15 |     #[delegate_to(x => &x.0)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot find attribute `delegate_to` in this scope
  --> tests/ui/delegate_to_fail_delegate_to_can_appear_at_most_once.rs:14:7
   |
14 |     #[delegate_to(x => &x.0)]
   |       ^^^^^^^^^^^

error: cannot find attribute `delegate_to` in this scope
  --> tests/ui/delegate_to_fail_delegate_to_can_appear_at_most_once.rs:15:7
   |
15 |     #[delegate_to(x => &x.0)]
   |       ^^^^^^^^^^^
